x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/version/EsqlVersionTests.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.version;
+
+import org.elasticsearch.test.ESTestCase;
+
+import java.util.Arrays;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+
+public class EsqlVersionTests extends ESTestCase {
+    public void testVersionString() {
+        assertThat(EsqlVersion.NIGHTLY.toString(), equalTo("nightly.\uD83D\uDE34"));
+        assertThat(EsqlVersion.PARTY_POPPER.toString(), equalTo("2024.04.\uD83C\uDF89"));
+    }
+
+    public void testVersionId() {
+        assertThat(EsqlVersion.NIGHTLY.id(), equalTo(Integer.MAX_VALUE));
+        assertThat(EsqlVersion.PARTY_POPPER.id(), equalTo(202404));
+    }
+
+    public void testVersionStringNoEmoji() {
+        for (EsqlVersion version : EsqlVersion.values()) {
+            String[] versionSegments = version.toString().split("\\.");
+            String[] parsingPrefixSegments = Arrays.copyOf(versionSegments, versionSegments.length - 1);
+
+            String expectedParsingPrefix = String.join(".", parsingPrefixSegments);
+            assertThat(version.versionStringNoEmoji(), equalTo(expectedParsingPrefix));
+        }
+    }
+
+    public void testParsing() {
+        for (EsqlVersion version : EsqlVersion.values()) {
+            int suffixLength = randomIntBetween(0, 10);
+            String validVersionString = version.versionStringNoEmoji() + randomUnicodeOfLength(suffixLength);
```

Comment:
ha, yes, this is exactly what I don't want to allow when I said we should be more defensive above.  I think we should accept exactly the date+emoji or exactly the date, and not accept date+anything else.